packages/shared/lib/clients/provider.client.ts

```diff
@@ -373,6 +388,66 @@ class ProviderClient {
         };
     }
 
+    private async createCorosToken(tokenUrl: string, code: string, clientId: string, clientSecret: string, callback: string): Promise<object> {
+        const headers = {
+            'Content-Type': 'application/x-www-form-urlencoded'
+        };
+        const body = {
+            client_id: clientId,
+            redirect_uri: callback,
+            code: code,
+            client_secret: clientSecret,
+            grant_type: 'authorization_code'
+        };
+        try {
+            const response = await axios.post(tokenUrl, body, { headers: headers });
+            if (response.status === 200 && response.data !== null) {
+                return {
+                    access_token: response.data['access_token'],
+                    refresh_token: response.data['refresh_token'],
+                    expires_in: response.data['expires_in'],
+                    openId: response.data['openId']
```

Comment:
should we make sure those attributes actually exists? maybe using zod validation? I know we are not doing it for the other ones but it is never too late to start. :D

Reply:
This file needs a full a refacto tbh not sure it's worth starting an initiative